src/main.c

```diff
@@ -162,21 +163,79 @@ struct {
   char parm_member[8+1];
   char *root_dir;
   char *workspace_dir;
-  
+  JsonArray *sys_messages;
   char ha_instance_id[64];
   
   pid_t pid;
   char userid[9];
   
 } zl_context = {.config = {.debug_mode = false}, .userid = "(NONE)"} ;
 
+static void set_sys_messages(ConfigManager *configmgr) {
+  Json *env;
+  int cfgGetStatus = cfgGetAnyC(configmgr, ZOWE_CONFIG_NAME, &env, 2, "zowe", "sysMessages");
 
+  if (cfgGetStatus != ZCFG_SUCCESS) { // No sysMessages found in Zowe configuration
+    return;
+  }
+  JsonArray *sys_messages = jsonAsArray(env);
+  
+  if (sys_messages) {
+    zl_context.sys_messages = sys_messages;
+  }
+}
 
-#define INFO(fmt, ...)  printf("%s <%s:%d> %s INFO "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)
-#define WARN(fmt, ...)  printf("%s <%s:%d> %s WARN "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)
-#define DEBUG(fmt, ...) if (zl_context.config.debug_mode) \
+static void check_for_and_print_sys_message(const char* fmt, ...) {
+  
+  if (!zl_context.sys_messages) {
+    return;
+  }
+  
+  /* All of this stuff here is because I can't do 
+  #define INFO(fmt, ...)  check_for_and_print_sys_message(fmt, ...) so let's make a string */
+  char input_string[1024];
+  va_list args;
+  va_start(args, fmt);
+  vsnprintf(input_string, sizeof(input_string), fmt, args);
+  va_end(args);
+  
+  /* Uncomment code to try to pull ID from input_string
+  // Extract the ID from input_string
+  char msg_id[256]; // assuming the ID will not exceed 255 characters
+  const char* spacePos = strchr(input_string, ' ');
+  if (spacePos) {
+      int length = spacePos - input_string;
+      strncpy(msg_id, input_string, length);
+      msg_id[length] = '\0';
+  } else {
+      // If no space found, use the whole input_string as the ID
+      //strncpy(msg_id, input_string, sizeof(msg_id) - 1);
+      //msg_id[sizeof(msg_id) - 1] = '\0'; // ensure null termination
+      
+      // If no space found, end
+      return;
+  } */
+    
+  int count = jsonArrayGetCount(zl_context.sys_messages);
+  for (int i = 0; i < count; i++) {
+      const char *sys_message_id = jsonArrayGetString(zl_context.sys_messages, i);
+      if (sys_message_id && strstr(input_string, sys_message_id)) {
+          wtoPrintf3(input_string); // Print our match to the syslog
+          break;
+      }
+  }
+  
+}
+
+#define INFO(fmt, ...)  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
+  printf("%s <%s:%d> %s INFO "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)
+#define WARN(fmt, ...)  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
+  printf("%s <%s:%d> %s WARN "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)
+#define DEBUG(fmt, ...) check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
```

Comment:
Maybe this part needs to be
`
if (zl_context.config.debug_mode) \
  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
  printf("%s <%s:%d> %s DEBUG "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)`